spec.bs

```diff
@@ -6520,11 +6541,52 @@ Each {{InterestGroupScriptRunnerGlobalScope}} has a
 <div algorithm>
 The <dfn attribute for=InterestGroupScriptRunnerGlobalScope>privateAggregation</dfn> [=getter
   steps=] are:
-
   1. Return [=this=]'s [=relevant global object=]'s
     [=InterestGroupScriptRunnerGlobalScope/private aggregation=].
 </div>
 
+<div algorithm>
+The <dfn attribute for=InterestGroupScriptRunnerGlobalScope>protectedAudience</dfn> [=getter
+  steps=] are:
+  1. Return [=this=]'s [=relevant global object=]'s
+    [=InterestGroupScriptRunnerGlobalScope/protected audience utilities=].
+</div>
+
+<div algorithm>
+The <dfn method for="ProtectedAudienceUtilities">encodeUtf8(USVString input)</dfn> method steps are:
+1. Follow the steps of {{TextEncoder/encode(input)|TextEncoder.encode(input)}}.
+
+</div>
+
+<div algorithm>
+The <dfn method for="ProtectedAudienceUtilities">decodeUtf8(Uint8Array bytes)</dfn> method steps are:
+1. Let |decoder| be a new instance of [=UTF-8=]'s [=decoder=].
+1. Let |ioQueue| be an <a for=/>I/O queue</a> of bytes « [=end-of-queue=] ».
+1. [=I/O queue/Push=] <var ignore>bytes</var> to |ioQueue|.
+1. Let |output| be an <a for=/>I/O queue</a> of bytes « [=end-of-queue=] ».
+1. While true:
+  1. Let |item| be the result of [=I/O queue/reading=] from |ioQueue|.
+  1. Let |result| be the result of [=processing an item=] with |item|, |decoder|, |ioQueue|,
+    |output|, and "<code>replacement</code>".
+  1. [=Assert=] that |result| is not an [=encoding-error|error=].
+  1. If |result| is [=encoding-finished|finished=], [=iteration/break=].
+1. Let |serializationSettings| be a new {{TextDecoderCommon}} with following fields:
+    <dl>
+      <dt>[=TextDecoderCommon-encoding|encoding=]
+      <dd>[=UTF-8=]
+      <dt>[=TextDecoderCommon-ignore BOM|ignore BOM=]
+      <dd>true
+      <dt>[=TextDecoderCommon-BOM seen|BOM seen=]
+      <dd>false
+    </dl>
+1. Return the result of running <a for=/>serialize I/O queue</a> with |serializationSettings|
+  and |output|.
+
+Note: This is a simplified version of {{TextDecoder/decode(input, options)|TextDecoder.decode(bytes)}},

```

Comment:
nit: probably clearer if we move this to the start of this algorithm?

Reply:
Done, as a non-note.